pkg/services/govmomi/service.go

```diff
@@ -37,11 +37,18 @@ import (
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/extra"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/net"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/task"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/util"
 )
 
+const (
+	maxTaskCount = 80
```

Comment:
we should make this value configurable, but given that the rate-limiting for now is a not concern a concern for the majority of the users it is not required to included this in this PR (it a low priority/backlog task)